ibis/pandas/execution/window.py

```diff
@@ -130,24 +132,64 @@ def get_aggcontext_window(
     return aggcontext
 
 
+def trim_with_timecontext(data, timecontext):
+    try:
+        begin, end = map(pd.to_datetime, timecontext)
+    except ValueError:
+        raise com.IbisError('Cannot resolve timecontext for window\n{}.')
+    else:
+        df = data.reset_index()
```

Comment:
The caller side seems to pass in pd.Series, but it's df inside?

Reply:
It is a Series for the input `data`, but once we reset the index it becomes a Dataframe.